app/assets/src/components/ui/controls/SearchBox.jsx

```diff
@@ -11,51 +12,64 @@ class SearchBox extends React.Component {
     this.waitHandleSearchChange = 500;
     this.minChars = 2;
 
-    this.source = this.props.source;
     this.placeholder = this.props.placeholder;
-
-    this.resetComponent = this.resetComponent.bind(this);
-    this.handleSearchChange = this.handleSearchChange.bind(this);
+    this.handleEnter = this.props.onEnter;
     this.handleResultSelect = this.handleResultSelect.bind(this);
-
     this.blankState = { isLoading: false, results: [], value: "" };
+
     this.state = {
       isLoading: false,
       results: [],
       value: this.props.initialValue
     };
   }
 
-  resetComponent() {
-    this.setState(this.blankState);
-  }
+  onKeyDown = e => {
```

Comment:
Why do we need onEnter? Isn't search automatic?

Reply:
Search is automatic but only triggered onSearchChange. If the input didn't change, pressing Enter won't do anything by default. Jenn said when you hit enter (as opposed to clicking one of the suggestions), the search should work just like it does now

Reply:
ok